langroid/embedding_models/models.py

```diff
@@ -139,6 +171,72 @@ def embedding_dims(self) -> int:
         return self.config.dims
 
 
+class AzureOpenAIEmbeddings(EmbeddingModel):
+    """
+    Azure OpenAI embeddings model implementation.
+    """
+
+    def __init__(
+        self, config: AzureOpenAIEmbeddingsConfig = AzureOpenAIEmbeddingsConfig()
+    ):
+        """
+        Initializes Azure OpenAI embeddings model.
+
+        Args:
+            config: Configuration for Azure OpenAI embeddings model.
+        Raises:
+            ValueError: If required Azure config values are not set.
+        """
+        super().__init__()
+        self.config = config
+        load_dotenv()
+
+        self.config.api_key = os.getenv("AZURE_OPENAI_API_KEY", "")
```

Comment:
can you please change it to:
```
    if not self.config.api_key:
        self.config.api_key = os.getenv("AZURE_OPENAI_API_KEY", "")
```
so that if user passes api_key in Config object it's not overwritten?